testsuite/MDAnalysisTests/coordinates/test_imd.py

```diff
@@ -0,0 +1,663 @@
+"""Test for MDAnalysis trajectory reader expectations
+"""
+import numpy as np
+import logging
+import pytest
+from MDAnalysis.transformations import translate
+import pickle
+import MDAnalysis as mda
+from numpy.testing import (
+    assert_almost_equal,
+    assert_array_almost_equal,
+    assert_equal,
+    assert_allclose,
+)
+from MDAnalysis.coordinates.IMD import HAS_IMDCLIENT
+
+if HAS_IMDCLIENT:
+    import imdclient
+    from imdclient.tests.utils import (
+        get_free_port,
+        create_default_imdsinfo_v3,
+    )
+    from imdclient.tests.server import InThreadIMDServer
+
+from MDAnalysis.coordinates.IMD import IMDReader
+
+from MDAnalysisTests.datafiles import (
+    COORDINATES_TOPOLOGY,
+    COORDINATES_TRR,
+    COORDINATES_H5MD,
+)
+
+from MDAnalysisTests.coordinates.base import (
+    MultiframeReaderTest,
+    BaseReference,
+    assert_timestep_almost_equal,
+)
+
+logger = logging.getLogger("imdclient.IMDClient")
+file_handler = logging.FileHandler("test.log")
+formatter = logging.Formatter(
+    "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+logger.setLevel(logging.DEBUG)
+
+@pytest.mark.skipif(not HAS_IMDCLIENT, reason="IMDClient not isntalled")
+class IMDReference(BaseReference):
+    def __init__(self):
+        super(IMDReference, self).__init__()
+        self.port = get_free_port()
+        # Serve TRR traj data via the server
+        traj = mda.coordinates.TRR.TRRReader(COORDINATES_TRR)
+        self.server = InThreadIMDServer(traj)
+        self.server.set_imdsessioninfo(create_default_imdsinfo_v3())
+
+        self.n_atoms = traj.n_atoms
+        self.prec = 3
+
+        self.trajectory = f"imd://localhost:{self.port}"
+        self.topology = COORDINATES_TOPOLOGY
+        self.changing_dimensions = True
+        self.reader = IMDReader
+
+        self.first_frame.velocities = self.first_frame.positions / 10
+        self.first_frame.forces = self.first_frame.positions / 100
+
+        self.second_frame.velocities = self.second_frame.positions / 10
+        self.second_frame.forces = self.second_frame.positions / 100
+
+        self.last_frame.velocities = self.last_frame.positions / 10
+        self.last_frame.forces = self.last_frame.positions / 100
+
+        self.jump_to_frame.velocities = self.jump_to_frame.positions / 10
+        self.jump_to_frame.forces = self.jump_to_frame.positions / 100
+
+    def iter_ts(self, i):
+        ts = self.first_frame.copy()
+        ts.positions = 2**i * self.first_frame.positions
+        ts.velocities = ts.positions / 10
+        ts.forces = ts.positions / 100
+        ts.time = i
+        ts.frame = i
+        return ts
+
+
+@pytest.mark.skipif(not HAS_IMDCLIENT, reason="IMDClient not isntalled")
+class TestIMDReaderBaseAPI(MultiframeReaderTest):
+
+    @pytest.fixture()
```

Comment:
```suggestion
    @pytest.fixture(scope='function')
```
If you need it at every test, then it's best to be explicit about the fixture scope